app/models/manageiq/providers/redhat/infra_manager/supported_apis_mixin.rb

```diff
@@ -0,0 +1,53 @@
+module ManageIQ::Providers::Redhat::InfraManager::SupportedApisMixin
+  def supported_api_versions
+    return supported_api_versions_from_sdk(probe_args) if api_version.blank?
+    supported_api_versions_from_db
+  end
+
+  def highest_supported_api_version
+    supported_api_versions.sort.last || '3'
+  end
+
+  def highest_allowed_api_version
+    return '3' unless use_ovirt_sdk?
+    highest_supported_api_version
+  end
+
+  # This method is a result of api_version in the db not actually being the api_version
+  # it is actually the product version based on which we infer the api version.
+  # In the future I hope there will be a separate field created.
+  def supported_api_versions_from_db
+    product_v_to_api_v_hash = Settings::ems::ems_redhat.default_product_version_to_api_version_regex_hash
```

Comment:
Instead of accessing directly the settings, I think we could have a method to read it, and maybe do some parsing of the values, to avoid the Ruby syntax.